app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java

```diff
@@ -79,154 +102,169 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
-//        mLocationRequest.setInterval(UPDATE_INTERVAL);
-//        mLocationRequest.setFastestInterval(FASTEST_INTERVAL);
-//
-//        LocationSettingsRequest.Builder builder = new LocationSettingsRequest.Builder();
-//        builder.addLocationRequest(mLocationRequest);
-//        LocationSettingsRequest locationSettingsRequest = builder.build();
-//
-//        SettingsClient settingsClient = LocationServices.getSettingsClient(this);
-//        settingsClient.checkLocationSettings(locationSettingsRequest);
-//        //noinspection MissingPermission
-//        getFusedLocationProviderClient(this).requestLocationUpdates(mLocationRequest, new LocationCallback() {
-//                    @Override
-//                    public void onLocationResult(LocationResult locationResult) {
-//                        onLocationChanged(locationResult.getLastLocation());
-//                    }
-//                },
-//                Looper.myLooper());
-//    }
-//
-//    public void onLocationChanged(Location location) {
-//        // GPS may be turned off
-//        if (location == null) {
-//            return;
-//        }
-//
-//        mCurrentLocation = location;
-//        String msg = "Updated Location: " +
-//                Double.toString(location.getLatitude()) + "," +
-//                Double.toString(location.getLongitude());
-//        //Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
-//        displayLocation();
-//    }
-//
-//    public void onSaveInstanceState(Bundle savedInstanceState) {
-//        savedInstanceState.putParcelable(KEY_LOCATION, mCurrentLocation);
-//        super.onSaveInstanceState(savedInstanceState);
-//    }
-//
-//    @Override
-//    public void onMapLongClick(@NonNull LatLng latLng) {
-//
-//    }
+        map = googleMap;
+        getLocation();
+//        googleMap.addMarker(new MarkerOptions()
+//                .position(new LatLng(51.5072, 0.1276))
+//                .title("London"));
+//        LatLng latLng = new LatLng(51.5072, 0.1276);
+//        map.moveCamera(CameraUpdateFactory.newLatLng(latLng));
+//        map.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng,15f));
+    }
+
+    private void getLocation() {
+        progressDialog.show();
+        if(Build.VERSION.SDK_INT>=23){
+            if(checkPermission()){
+                getDeviceLocation();
+            }
+            else{
+                requestPermission();
+            }
+        }
+        else{
+            getDeviceLocation();
+        }
+    }
+
+    private void requestPermission() {
+        ActivityCompat.requestPermissions(getActivity(),permissions_all,PERMISSION_CODE);
+    }
+
+    private boolean checkPermission() {
+        for(int i=0;i<permissions_all.length;i++){
+            int result= ContextCompat.checkSelfPermission(getContext(),permissions_all[i]);
+            if(result== PackageManager.PERMISSION_GRANTED){
+                continue;
```

Comment:
Its been my experience that continues are confusing, especially when you nest for loops, so to protect yourself against potential bugs, its a good idea to refactor this so the continue isn't necessary. For example, here, you can just write "if (result != Permission_granted) { return false; }" which is a bit cleaner and does the same thing.